src/main/scala/io/github/sentenza/hacktoberfest18/algos/ImmutableSelection.scala

```diff
@@ -0,0 +1,52 @@
+package io.github.sentenza.hacktoberfest18.algos
+
+import scala.collection.mutable.ArrayBuffer
+import scala.math.Ordered
+
+/*
+ * HacktoberFest 2018 - Scala Algorhitms
+ * Copyright (C) 2018 sentenza
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+  * Implementations of this trait should provide non-destructive select
+  * operations on array. That is, the array passed to a select method
+  * should not be altered: A value at the specified sorted index should be
+  * returned.
+  */
+object ImmutableSelection extends Selection[Array, Int] {
```

Comment:
The `ImmutableSelection` object must use immutable collections, e.g. `List`:

```suggestion
object ImmutableSelection extends Selection[List, Int] {
```